mobile/library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -322,12 +322,12 @@ String createYaml() {
 
     for (StackTraceElement element : Thread.currentThread().getStackTrace()) {
       if (element.getClassName().startsWith("org.junit.")) {
-        List<EnvoyNativeFilterConfig> reverseFilterChain = nativeFilterChain;
+        List<EnvoyNativeFilterConfig> reverseFilterChain = new ArrayList<>(nativeFilterChain);
         Collections.reverse(reverseFilterChain);
         Boolean enforceTrustChainVerification =
             trustChainVerification == EnvoyConfiguration.TrustChainVerification.VERIFY_TRUST_CHAIN;
 
-        byte[][] filter_chain = JniBridgeUtility.toJniBytes(nativeFilterChain);
+        byte[][] filter_chain = JniBridgeUtility.toJniBytes(reverseFilterChain);
```

Comment:
drive-by: This looks like a prod-bug not a test-bug. However, the PR description mentions "junit" which (I think) is a test thing. Should we tweak the PR description? (Should we have a test for this)

Reply:
this is inside 

for (StackTraceElement element : Thread.currentThread().getStackTrace()) {
      if (element.getClassName().startsWith("org.junit.")) {

we're about to have a test as it broke something in the "use the bootstrap from Java" PR but I can land a separate one as well as a follow-up

Reply:
Oooooh! Sorry, good point!